File: web/lite/src/codec/tags/solid-layer.ts
import { SolidLayer } from '../../base/solid-layer';
import { readColor } from '../data-types';
import { ByteArray } from '../utils/byte-array';

export function readSolidColor(byteArray: ByteArray, layer: SolidLayer) {
  layer.solidColor = readColor(byteArray);
  layer.width = byteArray.readEncodeInt32();
  layer.height = byteArray.readEncodeInt32();
}

File: src/base/SolidLayer.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Tencent is pleased to support the open source community by making libpag available.
//
//  Copyright (C) 2021 Tencent. All rights reserved.
//
//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
//  except in compliance with the License. You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  unless required by applicable law or agreed to in writing, software distributed under the
//  license is distributed on an "as is" basis, without warranties or conditions of any kind,
//  either express or implied. see the license for the specific language governing permissions
//  and limitations under the license.
//
/////////////////////////////////////////////////////////////////////////////////////////////////

#include "base/utils/Verify.h"
#include "pag/file.h"

namespace pag {
bool SolidLayer::verify() const {
  if (!Layer::verify()) {
    VerifyFailed();
    return false;
  }
  VerifyAndReturn(width > 0 && height > 0);
}

Rect SolidLayer::getBounds() const {
  return Rect::MakeWH(width, height);
}
}  // namespace pag
